src/misc.cpp

```diff
@@ -452,7 +437,20 @@ void* aligned_ttmem_alloc(size_t allocSize) {
 #else
 
 void* aligned_ttmem_alloc(size_t allocSize) {
-  return std_aligned_alloc(4096, allocSize);
+
+#if defined(__linux__)
+  constexpr size_t alignment = 2 * 1024 * 1024; // assumed 2MB page size
```

Comment:
Enabling this on Android was not an omission by me. Android devices actually have transparent huge pages enabled (at least in the ones I've inspected), so aligning the hash/weights by 2MB makes sense, even if we don't have madvise() available. The devices I've seen used THP in always/always mode, so all they need is alignment.  

However, THP is actually somewhat important in portableAndroid devices, since perf increase = more power saving for fixed-size bulk tasks. I've no idea why bionic (the Android C lib) doesn't expose madvise(), though.

Reply:
Yes, strange, since I just did a quick search, and found that the kernel actually seems to support this, but that doesn't mean it is exposed however: https://android.googlesource.com/kernel/msm/+/android-msm-marlin-3.18-nougat-dr1/mm/madvise.c